tembo-operator/src/dedicated_networking.rs

```diff
@@ -0,0 +1,575 @@
+                    e
+                })?;
+
+            debug!(
+                "Reconciling IP allow list middleware for CoreDB instance: {}",
+                cdb.name_any()
+            );
+            let middleware_name = reconcile_ip_allowlist_middleware(cdb, ctx.clone())
+                .await
+                .map_err(|e| {
+                    error!("Failed to reconcile IP allowlist middleware: {:?}", e);
+                    e
+                })?;
+
+            debug!(
+                "Handling primary service ingress for CoreDB instance: {}",
+                cdb.name_any()
+            );
+            reconcile_dedicated_networking_service(
+                cdb,
+                client.clone(),
+                &ns,
+                dedicated_networking.public,
+                false,
+                &dedicated_networking.serviceType,
+            )
+            .await
+            .map_err(|e| {
+                error!("Failed to reconcile primary service ingress: {:?}", e);
+                e
+            })?;
+
+            reconcile_dedicated_networking_ing_route_tcp(
+                cdb,
+                ctx.clone(),
+                &ns,
+                basedomain,
+                "dedicated",
+                &format!("{}-dedicated", cdb.name_any()),
+                port.clone(),
+                vec![middleware_name.clone()],
+                false,
+                false,
+            )
+            .await
+            .map_err(|e| {
+                error!("Failed to reconcile primary ingress route: {:?}", e);
+                e
+            })?;
+
+            if dedicated_networking.includeStandby {
+                info!(
+                    "Handling standby service ingress for CoreDB instance: {}",
+                    cdb.name_any()
+                );
+                reconcile_dedicated_networking_service(
+                    cdb,
+                    client.clone(),
+                    &ns,
+                    dedicated_networking.public,
+                    true,
+                    &dedicated_networking.serviceType,
+                )
+                .await
+                .map_err(|e| {
+                    error!("Failed to reconcile standby service ingress: {:?}", e);
+                    e
+                })?;
+
+                reconcile_dedicated_networking_ing_route_tcp(
+                    cdb,
+                    ctx.clone(),
+                    &ns,
+                    basedomain,
+                    "dedicated-ro",
+                    &format!("{}-dedicated-ro", cdb.name_any()),
+                    port.clone(),
+                    vec![middleware_name],
+                    false,
+                    true,
+                )
+                .await
+                .map_err(|e| {
+                    error!("Failed to reconcile standby ingress route: {:?}", e);
+                    e
+                })?;
+            }
```

Comment:
If standby is disabled, it should delete the standby service. For example - enable HA, enable dedicated networking, disable HA. Should end up with 1 dedicated service only